### PR TITLE
Add MySQL 5.7.10 support

### PR DESCRIPTION
**Pull request by liucijus**
tested only on Linux (Ubuntu 14.04.3 LTS)

### CODE REVIEW

File: src/main/java/com/wix/mysql/MysqldExecutable.java [hunk 1]
```diff
@@ -73,16 +71,40 @@ private void initDatabase() throws IOException {
         }
     }
 
+    private Process getProcessForVersion57() throws IOException {
+        return Runtime.getRuntime().exec(new String[]{
+                        "bin/mysqld",
+                        "--no-defaults",
+                        "--initialize-insecure",
+                        "--user=vaidas",
```
Comment (viliusl) [inline src/main/java/com/wix/mysql/MysqldExecutable.java]: :)

File: src/main/java/com/wix/mysql/MysqldProcess.java [hunk 2]
```diff
@@ -84,6 +82,11 @@ public void onAfterProcessStart(final ProcessControl process, final IRuntimeConf
                 format("--character-set-server=%s", config.getCharset().getCharset()),
                 format("--collation-server=%s", config.getCharset().getCollate()),
                 format("--default-time-zone=%s", Utils.asHHmmOffset(config.getTimeZone())));
+
+        if (!config.getVersion().getMajorVersion().equals("5.7"))
+            commandLine.add("--skip-name-resolve");
```
Comment (viliusl) [inline src/main/java/com/wix/mysql/MysqldProcess.java]: maybe we can drop this altogether?

**Comment by viliusl** [comment]
So now I will have to fire-up my windows vm and test it:/

Otherwise nice job! Will have some fun stuff to merge - guava, 5.7 support, just need to fix that Travis thingie first...

File: src/main/java/com/wix/mysql/MysqldExecutable.java (hunk 1 above)
Comment (liucijus) [inline src/main/java/com/wix/mysql/MysqldExecutable.java]: oops

File: src/main/java/com/wix/mysql/MysqldProcess.java (hunk 2 above)
Comment (liucijus) [inline src/main/java/com/wix/mysql/MysqldProcess.java]: this switch is required for versions prior to 5.7

File: src/main/java/com/wix/mysql/MysqldExecutable.java [hunk 3]
```diff
@@ -41,26 +41,24 @@ protected MysqldProcess start(
...
-            Process p = Runtime.getRuntime().exec(new String[]{
-                            "scripts/mysql_install_db",
-                            "--no-defaults",
-                            format("--basedir=%s", getBaseDir()),
-                            format("--datadir=%s/data", getBaseDir())},
-                    null,
-                    getBaseDir());
+            boolean isSeverVersion57 = config.getVersion().getMajorVersion().equals("5.7");
+
+            System.out.println(getBaseDir());
```
Comment (noam-almog) [inline src/main/java/com/wix/mysql/MysqldExecutable.java]: debug line, remove it please

Comment (liucijus) [inline src/main/java/com/wix/mysql/MysqldExecutable.java]: thanks for catching

**Comment by viliusl** [comment]
Hej, travis ir more or less ok, so I will be able to merge it, but please merge with master and include https://github.com/wix/wix-embedded-mysql/commit/3edfe1265507e624df73bdf6ac954b6defd4e3fa which fixes os x issues.

Windows is going to be tuff, as they changed alot in 5.7, but I think I will merge without windows support and will add it eventually as I will need to do some refactoring within codebase.

**Pull request merged**